include/boost/basic_any.hpp

```diff
@@ -0,0 +1,440 @@
+                    break;
+                case Move:
+                    new (&left.content.small_value) ValueType(std::move(*reinterpret_cast<ValueType*>(&const_cast<basic_any*>(right)->content.small_value)));
+                    left.man = right->man;
+                    reinterpret_cast<ValueType const*>(&right->content.small_value)->~ValueType();
+                    const_cast<basic_any*>(right)->man = 0;
+                    break;
+                case Copy:
+                    new (&left.content.small_value) ValueType(*reinterpret_cast<const ValueType*>(&right->content.small_value));
+                    left.man = right->man;
+                    break;
+                case AnyCast:
+                    return boost::typeindex::type_id<ValueType>() == *info ?
+                            reinterpret_cast<BOOST_DEDUCED_TYPENAME remove_cv<ValueType>::type *>(&left.content.small_value) : 0;
+                case UnsafeCast:
+                    return reinterpret_cast<BOOST_DEDUCED_TYPENAME remove_cv<ValueType>::type *>(&left.content.small_value);
+                case Typeinfo:
+                    return const_cast<void*>(static_cast<const void*>(&boost::typeindex::type_id<ValueType>().type_info()));
+            }
+
+            return 0;
+        }
+
+        template <typename ValueType>
+        static void* large_manager(operation op, basic_any& left, const basic_any* right, const boost::typeindex::type_info* info)
+        {
+            switch (op)
+            {
+                case Destroy:
+                    delete static_cast<ValueType*>(left.content.large_value);
+                    break;
+                case Move:
+                    left.content.large_value = right->content.large_value;
+                    left.man = right->man;
+                    const_cast<basic_any*>(right)->content.large_value = 0;
+                    const_cast<basic_any*>(right)->man = 0;
+                    break;
+                case Copy:
+                    left.content.large_value = new ValueType(*static_cast<const ValueType*>(right->content.large_value));
+                    left.man = right->man;
+                    break;
+                case AnyCast:
+                    return boost::typeindex::type_id<ValueType>() == *info ?
+                            static_cast<BOOST_DEDUCED_TYPENAME remove_cv<ValueType>::type *>(left.content.large_value) : 0;
+                case UnsafeCast:
+                    return reinterpret_cast<BOOST_DEDUCED_TYPENAME remove_cv<ValueType>::type *>(left.content.large_value);
+                case Typeinfo:
+                    return const_cast<void*>(static_cast<const void*>(&boost::typeindex::type_id<ValueType>().type_info()));
+            }
+
+            return 0;
+        }
+
+
+        template <typename ValueType>
+        struct is_small_object_impl : boost::integral_constant<bool, sizeof(ValueType) <= OptimizeForSize &&
```

Comment:
Drop the is_small_object down below. Rename is_small_object_impl to is_small_object

Reply:
Done